tools/cli/NearObjectCli.cxx

```diff
@@ -105,27 +105,24 @@ NearObjectCli::CreateParser() noexcept
 namespace detail
 {
 /**
- * @brief Helper function to produce a mapping of enumeration strings to their
- * corresponding field names.
- *
- * This is used for CLI11 CheckedTransformer which validates string inputs and
- * maps them to their corresponding enumeration value.
+ * @brief Helper function to produce a mapping of enumeration field names to their
+ * corresponding values.
  *
  * @tparam EnumType The type of the enumeration.
- * @return std::map<std::string, EnumType>
+ * @return std::map<std::string, int>
  */
 template <typename EnumType>
 // clang-format off
 requires std::is_enum_v<EnumType>
-const std::unordered_map<std::string, EnumType>
+const std::unordered_map<std::string, int>
 CreateEnumerationStringMap() noexcept
 // clang-format on
 {
     const auto& reverseMap = magic_enum::enum_entries<EnumType>();
-    std::unordered_map<std::string, EnumType> map;
+    std::unordered_map<std::string, int> map;
 
     for (const auto& [enumValue, enumName] : reverseMap) {
-        map.emplace(enumName, enumValue);
+        map.emplace(enumName, static_cast<int>(enumValue));
```

Comment:
Why do the casting here, instead of where the map is used? Casting an enum value to an `int` will truncate data when `sizeof(std::underlying_type_t<EnumType>) > sizeof(int)`. I'd much rather see this function kept as-is and avoid this cast since it's generic, then just cast in the loop in `AddEnumOption()` if determined to be safe.